rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/SchedulerClient.java

```diff
@@ -927,4 +938,77 @@ public void removeThirdPartyCredential(String key) throws NotConnectedException,
+            TaskInfoData taskInfo = taskStateData.getTaskInfo();
+            TaskIdData taskIdData = taskInfo.getTaskId();
+            JobId jobId = new JobIdImpl(taskInfo.getJobId().getId(), taskInfo.getJobId().getReadableName());
+            TaskId taskId = TaskIdImpl.createTaskId(jobId, taskIdData.getReadableName(), taskIdData.getId());
+            lTaskIds.add(taskId);
+        }
+        return new Page<TaskId>(lTaskIds, page.getSize());
+    }
+
+    @Override
+    public Page<TaskState> getTaskStates(String taskTag, long from, long to, boolean mytasks,
+            boolean running, boolean pending, boolean finished, int offset, int limit)
+                    throws NotConnectedException, PermissionException {
+        RestPage<TaskStateData> page = null;
+        try {
+            page = restApi().getTaskStates(sid, from, to, mytasks, running, pending, finished, offset,
+                    limit);
+        } catch (NotConnectedRestException e) {
+            throw new NotConnectedException(e);
+        } catch (PermissionRestException e) {
+            throw new PermissionException(e);
+        }
+        List<TaskState> lTaskStates = new ArrayList<TaskState>(page.getList().size());
+        for (TaskStateData taskStateData : page.getList()) {
+            lTaskStates.add(new TaskStateImpl(taskStateData));
+        }
+        return new Page<TaskState>(lTaskStates, page.getSize());
+    }
+
+    @Override
+    public JobInfo getJobInfo(String jobId)
+            throws UnknownJobException, NotConnectedException, PermissionException {
+        JobInfoData jobInfoData = null;
+        try {
+            jobInfoData = restApi().jobInfo(sid, jobId);
+        } catch (NotConnectedRestException e) {
+            throw new NotConnectedException(e);
+        } catch (PermissionRestException e) {
+            throw new PermissionException(e);
+        } catch (UnknownJobRestException e) {
+            throw new UnknownJobException(e);
+        }
+        JobInfoImpl jobInfoImpl = new JobInfoImpl();
+        JobId newJobId = JobIdImpl.makeJobId(jobId);
+        jobInfoImpl.setJobId(newJobId);
+        jobInfoImpl.setJobOwner(jobInfoData.getJobOwner());
+        jobInfoImpl.setFinishedTime(jobInfoData.getFinishedTime());
+        jobInfoImpl.setRemovedTime(jobInfoData.getRemovedTime());
+        jobInfoImpl.setStartTime(jobInfoData.getStartTime());
+        jobInfoImpl.setSubmittedTime(jobInfoData.getSubmittedTime());
+        jobInfoImpl.setNumberOfFinishedTasks(jobInfoData.getNumberOfFinishedTasks());
+        jobInfoImpl.setNumberOfPendingTasks(jobInfoData.getNumberOfPendingTasks());
+        jobInfoImpl.setNumberOfRunningTasks(jobInfoData.getNumberOfRunningTasks());
+        jobInfoImpl.setJobPriority(JobPriority.findPriority(jobInfoData.getPriority().toString()));
+        jobInfoImpl.setJobStatus(JobStatus.findPriority(jobInfoData.getStatus().toString()));
+        if (jobInfoData.isToBeRemoved()) jobInfoImpl.setToBeRemoved();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) At most one statement is allowed per line, but 2 statements were found on this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00122)